src/app/admin/collectionCentresTable/CollectionCentresTable.tsx

```diff
@@ -99,6 +155,62 @@ const CollectionCentresTable: React.FC = () => {
         };
     }, [getCollectionCentresForTable]);
 
+    const handleModalSaveClick = async (): Promise<void> => {
+        if (timeSlotModalData === null) {
+            return;
+        }
+        const { error: updateTimeSlotError } =
+            await updateDbCollectionCentreTimeSlots(timeSlotModalData);
+        const baseAuditLog = getBaseAuditLogForCollectionCentreTimeSlots(
+            "update collection centre time slots",
+            timeSlotModalData
+        );
+
+        if (updateTimeSlotError) {
+            setTimeSlotModalErrorMessage(
+                `Failed to update the collection centre time slots. Log ID: ${updateTimeSlotError.logId}`
+            );
+            void sendAuditLog({
+                ...baseAuditLog,
+                wasSuccess: false,
+                logId: updateTimeSlotError.logId,
+            });
+        }
+
+        void sendAuditLog({ ...baseAuditLog, wasSuccess: true });
+        setTimeSlotModalIsOpen(false);
+    };
+
+    const handleTimeSlotCheckBoxChange = (event: React.SyntheticEvent<Element, Event>): void => {
+        if (!timeSlotModalData) {
+            return;
+        }
+
+        const changedTimeSlotsData = timeSlotModalData.timeSlots.map((timeSlot) => {
+            if (
+                formatTimeStringToHoursAndMinutes(timeSlot.time) ===
+                event.currentTarget.parentElement?.parentElement?.innerText
+            ) {
+                return {
+                    time: timeSlot.time,
+                    isActive: (timeSlot.isActive = !timeSlot.isActive),
+                };
+            } else {
+                return {
+                    time: timeSlot.time,
+                    isActive: timeSlot.isActive,
+                };
+            }
+        });
```

Comment:
I think it might be more efficient to find the correct entry and update the value of that in place rather than updating the whole list?
```
const updatedTime = event.currentTarget.parentElement?.parentElement?.innerText;
let timeslot = timeSlotModalData.timeSlots.find(t => t.time === updatedTime);
timeslot.isActive = !timeslot.isActive;
```